lib/appcanary/configuration.rb

```diff
@@ -0,0 +1,51 @@
+require "yaml"
+
+module Appcanary
+  APPCANARY_DEFAULT_BASE_URI = "https://appcanary.com/api/v3/monitors"
+
+  class Configuration
+    attr_accessor :base_uri, :api_token, :monitor_name
+
+    def initialize
+      self.base_uri = APPCANARY_DEFAULT_BASE_URI
+    end
+  end
+
+  class ConfigurationError < RuntimeError
+  end
+
+  class << self
+    def configuration
+      @@configuration ||= Configuration.new
```

Comment:
Yeesh, the semantics of class level class/instance variables always escapes me. Thar be dragons.